pkg/storage/s3/checker.go

```diff
@@ -23,7 +24,7 @@ func (s *Storage) Exists(ctx context.Context, module, version string) (bool, err
 
 	lsParams := &s3.ListObjectsInput{
 		Bucket: aws.String(s.bucket),
-		Prefix: aws.String(config.PackageVersionedName(module, version, "")),
+		Prefix: aws.String(path.Dir(config.PackageVersionedName(module, version, ""))),
```

Comment:
I think you can just use `config.FmtModVer` instead of taking path.Dir(config.PackageVersionedName(...)).

Reply:
`config.FmtModVer` would return something like `github.com/golang/mock@1.4.4`, but we need `github.com/golang/mock/@v`

Reply:
@linzhp in that case adding a version then removing just to use existing helpers seems a little unnecessary. What if we just did this: `aws.String(fmt.Sprintf("%s/@v", module))`